lib/ansible/modules/identity/ipa/ipa_dnszone.py

```diff
@@ -0,0 +1,185 @@
+    - This only applies if C(ipa_prot) is I(https).
+    - If set to C(no), the SSL certificates will not be validated.
+    - This should only set to C(no) used on personally controlled sites using self-signed certificates.
+    required: false
+    default: true
+version_added: "2.4"
+'''
+
+EXAMPLES = '''
+# Ensure dns zone is present
+- ipa_dnsrecord:
+    ipa_host: spider.example.com
+    ipa_pass: Passw0rd!
+    state: present
+    zone_name: example.com
+
+# Ensure that dns zone is removed
+- ipa_dnszone:
+    zone_name: example.com
+    ipa_host: ipa.example.com
+    ipa_user: admin
+    ipa_pass: topsecret
+    state: absent
+'''
+
+RETURN = '''
+zone:
+  description: DNS zone as returned by IPA API.
+  returned: always
+  type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ipa import IPAClient
+from ansible.module_utils._text import to_native
+
+
+class DNSZoneIPAClient(IPAClient):
+    def __init__(self, module, host, port, protocol):
+        super(DNSZoneIPAClient, self).__init__(module, host, port, protocol)
+
+    def dnszone_find(self, zone_name):
+        return self._post_json(
+            method='dnszone_find',
+            name=zone_name,
+            item={'idnsname': zone_name}
+        )
+
+    def dnszone_add(self, zone_name=None, details=None):
+        return self._post_json(
+            method='dnszone_add',
+            name=zone_name,
+            item={}
+        )
+
+    def dnszone_del(self, zone_name=None, record_name=None, details=None):
+        return self._post_json(
+            method='dnszone_del', name=zone_name, item={})
+
+
+def ensure(module, client):
+    zone_name = module.params['zone_name']
+    state = module.params['state']
+
+    ipa_dnszone = client.dnszone_find(zone_name)
+
+    changed = False
+    if state == 'present':
+        if not ipa_dnszone:
+            changed = True
+            if not module.check_mode:
+                client.dnszone_add(zone_name=zone_name)
+        else:
+            changed = False
+    else:
+        if ipa_dnszone:
+            changed = True
+            if not module.check_mode:
+                client.dnszone_del(zone_name=zone_name)
+
+    return changed, client.dnszone_find(zone_name)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            zone_name=dict(type='str', required=True),
+            ipa_prot=dict(
+                type='str',
+                required=False,
+                default='https',
+                choices=['http', 'https']
+            ),
+            ipa_host=dict(
+                type='str',
+                required=False,
+                default='ipa.example.com'
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            ),
+            ipa_port=dict(type='int', required=False, default=443),
```

Comment:
One small nit pick: required by default is False so no need to specify for false values.